brave/browser-laptop
# Adding additional data clearing action for cookies only (Extension data no longer removed with clear cookies action)

Fixes Issue #13098

Continuation of #13110

## Submitter Checklist:


## Test Plan:

Test plan specified in issue:

1. Under Clear Browsing Data, and under Private Data in about:preferences#security, test the `Clear all stored site data` and `Clear cookies` option independently.
2. Ensure that only cookies are deleted, and extension data such as last pass credentials are not cleared out when selecting `Clear Cookies`
3. Ensure that all storage types for every session partition are cleared when selecting `Clear all stored site data`

## Reviewer Checklist:


Tests



Previously, there was one very general method being called when "clear all site cookies" was enabled called clearStorageData(). This deleted: appcache, cookies, filesystem, indexdb, local storage,shadercache, websql, and serviceworkers. This was way too aggressive. 

Per https://github.com/brave/muon/blob/master/atom/browser/api/atom_api_session.cc#L84

We can see that specific storage types can be specified, the only one we should need to touch for this option is "cookies"

In the filtering module, specific clearCache() and clearHistory() methods exist. I created an pseudo alias of the clearStorageData() method called clearCookies() that is called instead. bound to the clearStorageData() call is `{'storages': ['cookies']}` which specifies the cookies storage type.

See: https://github.com/electron/electron/blob/master/docs/api/session.md#sesclearstoragedataoptions-callback
